rasa/core/processor.py

```diff
@@ -325,21 +327,46 @@ def _has_message_after_reminder(
         ):
             logger.debug(
                 f"Canceled reminder because it is outdated. "
-                f"(event: {reminder_event.action_name} id: {reminder_event.name})"
+                f"(event: {reminder_event.intent} id: {reminder_event.name})"
             )
         else:
-            # necessary for proper featurization, otherwise the previous
-            # unrelated message would influence featurization
-            tracker.update(UserUttered.empty())
-            action = self._get_action(reminder_event.action_name)
-            should_continue = await self._run_action(
-                action, tracker, output_channel, nlg
+            intent = reminder_event.intent
+            entities = reminder_event.entities if reminder_event.entities else {}
+            await self.trigger_external_user_uttered(
+                intent, entities, tracker, output_channel
             )
-            if should_continue:
-                user_msg = UserMessage(None, output_channel, sender_id)
-                await self._predict_and_execute_next_action(user_msg, tracker)
-            # save tracker state to continue conversation from this state
-            self._save_tracker(tracker)
+
+    async def trigger_external_user_uttered(
+        self,
+        intent_name: Text,
+        entities: Union[List[Dict[Text, Any]], Dict[Text, Text]],
+        tracker: DialogueStateTracker,
+        output_channel: OutputChannel,
+    ) -> None:
+        """
+        Trigger an external message (like a user message,
+        but invisible; used, e.g., by a Reminder)
```

Comment:
I think it would be even better to add a proper docstring including the parameter descriptions.
```suggestion
        but invisible; used, e.g., by a Reminder).
```